pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/LuceneIndexScanDocIdIterator.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.operator.dociditerators;
+
+import org.apache.lucene.document.Document;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.pinot.core.common.Constants;
+import org.apache.pinot.core.segment.creator.impl.V1Constants;
+import org.apache.pinot.core.segment.index.readers.text.LuceneIndexSearcherReferenceManager;
+import org.apache.pinot.core.segment.index.readers.text.LuceneTextIndexReader;
+import org.roaringbitmap.IntIterator;
+
+
+public class LuceneIndexScanDocIdIterator implements IndexBasedDocIdIterator {
+
+  private int _currentDocId = -1;
+  private int _startDocId;
+  private int _endDocId;
+  private final IntIterator _docIDIterator;
+  private final IndexSearcher _indexSearcher;
+  private final LuceneIndexSearcherReferenceManager _searcherReferenceManager;
+  private final int _numDocs;
+
+  public LuceneIndexScanDocIdIterator(LuceneTextIndexReader.LuceneSearchResult luceneSearchResult) {
+    _docIDIterator = luceneSearchResult.getDocIDs().getIntIterator();
+    _numDocs = luceneSearchResult.getDocIDs().getCardinality();
+    _searcherReferenceManager = luceneSearchResult.getIndexSearcherReferenceManager();
+    _indexSearcher = _searcherReferenceManager.getIndexSearcher();
+  }
+
+  public void setStartDocId(int startDocId) {
+    _startDocId = startDocId;
+  }
+
+  public void setEndDocId(int endDocId) {
+    _endDocId = endDocId;
+  }
+
+  @Override
+  public int next() {
+    if (_currentDocId == Constants.EOF || !_docIDIterator.hasNext()) {
```

Comment:
nit _docId

Reply:
No longer needed since this file is removed.